src/integrations/prefect-dask/tests/test_task_runners.py

```diff
@@ -248,6 +249,25 @@ def test_flow():
         assert bx.type == StateType.PENDING
         assert cx.type == StateType.COMPLETED
 
+    def test_as_completed_yields_correct_order(self, task_runner):
+        @task
+        def sleep_task(seconds):
+            time.sleep(seconds)
+            return seconds
+
+        timings = [1, 5, 10]
+        with task_runner:
+            done_futures = []
+            futures = [
+                task_runner.submit(
+                    sleep_task, parameters={"seconds": seconds}, wait_for=[]
+                )
+                for seconds in timings
```

Comment:
I think this test would be more compelling if the timings were ordered in reverse, proving that `as_completed` actually affected the order that they yielded.
```suggestion
                for seconds in reversed(timings)
```